temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -952,6 +939,42 @@ public Functions.Proc scheduleLocalActivityTask(
   private void validateCommand(Command command, HistoryEvent event) {
     // TODO(maxim): Add more thorough validation logic. For example check if activity IDs are
     // matching.
+
+    // ProtocolMessageCommand is different from other commands because it can be associated with
+    // multiple types of events
+    if (command.getCommandType() == COMMAND_TYPE_PROTOCOL_MESSAGE) {
+      ProtocolMessageCommandAttributes commandAttributes =
+          command.getProtocolMessageCommandAttributes();
+      switch (event.getEventType()) {
+        case EVENT_TYPE_WORKFLOW_EXECUTION_UPDATE_ACCEPTED:
+          assertMatch(
+              command,
+              event,
+              "messageType",
+              true,
+              commandAttributes.getMessageId().endsWith("accept"));
```

Comment:
I don't love it but I get it. Is it possible to add the message set to the args to validateCommand? Not blocking but it would be better to test the actual message type.

Reply:
+1 to that

Reply:
I'll create a follow up issue to improve